pywbem_mock/_wbemconnection_mock.py

```diff
@@ -449,7 +450,117 @@ def add_cimobjects(self, objects, namespace=None):
                 except KeyError:
                     self.qualifiers[namespace] = NocaseDict({qual.name: qual})
             else:
-                assert False, 'add_cimobjects. %s not valid.' % type(obj)
+                assert False, 'Object to add_cimobjects. %s invalid type' \
+                              % type(obj)
+
+    def subscribe_method(self, classname, methodname, method_callback,
+                         namespace=None,):
+        """
+        Add a callback for a method that will be called by the client when
+        InvokeMethod is executed.  This method does no real validation of
+        the classname, methodname or method_callback in put parameters.
+
+          Parameters:
+            classname (:term:`string`):
+                The classname for the method defined by methodname.
+
+            methodname (:term: `string`):
+                Name of the method to be executed.
+
+            method_callback (:term:`callable`):
+                The user-defined function or method that well be called when by
+                FakeWBEMConnection when the InvokeMethod for the defined
+                class and methodname is received.  This method will be called
+                with the parameters defined on the InvokeMethod call
+
+                The callback method defined by the user is expected to comply
```

Comment:
It is possible to define a global function that defines and describes the callback interface. That function can then be referenced here, and the type of the method_callback parameter would be the function name.

An example of how that is done, is in `pywbem/_listener.py`, where `callback_interface()` defines the interface of the callback function and `WBEMListener.add_callback()` adds a callback function.

Reply:
Agreed. See the listener as an example